packages/editor/src/components/browser/Browser.tsx

```diff
@@ -46,8 +48,13 @@ const Browser = ({ open, onOpenChange, types, accept, location, cmsOptions }: Br
 
   const tabs = allBrowsers.filter(browser => types.includes(browser.id));
 
+  const handleTextSelection = () => {
+    const newSelectedText = window.getSelection()?.toString();
+    setSelectedText(newSelectedText === undefined ? '' : newSelectedText);
+  };
+
```

Comment:
I think it should be possible to reuse the logic for the replacement for this selected text...

Reply:
I have now adapted the entire thing so that a new function called "getMonacoSelection" is available in useModifyEditor, which returns the value of the current selection. I then pass this method to the browser, and it calls it when the browser icon is clicked. I have added the functionality only for ScriptArea fields for now.